bridges/FurAffinityBridge.php

```diff
@@ -0,0 +1,914 @@
+				'type-poetry' => ($this->getInput('type-poetry') === true ? 'on' : 0),
+				'mode' => $this->getInput('mode')
+			);
+			$html = $this->postFASimpleHTMLDOM($data);
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		case 'Browse':
+			$data = array(
+				'cat' => $this->getInput('cat'),
+				'atype' => $this->getInput('atype'),
+				'species' => $this->getInput('species'),
+				'gender' => $this->getInput('gender'),
+				'perpage' => 72,
+				'rating_general' => ($this->getInput('rating_general') === true ? 'on' : 0),
+				'rating_mature' => ($this->getInput('rating_mature') === true ? 'on' : 0),
+				'rating_adult' => ($this->getInput('rating_adult') === true ? 'on' : 0)
+			);
+			$html = $this->postFASimpleHTMLDOM($data);
+			$limit = (is_int($this->getInput('limit-browse')) ? $this->getInput('limit-browse') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		case 'Journals':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : -1);
+			$this->itemsFromJournalList($html, $limit);
+			break;
+		case 'Single Journal':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$this->itemsFromJournal($html);
+			break;
+		case 'Gallery':
+		case 'Scraps':
+		case 'Favorites':
+		case 'Gallery Folder':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		}
+	}
+
+	private function postFASimpleHTMLDOM($data) {
+			$opts = array(
+				CURLOPT_CUSTOMREQUEST => 'POST',
+				CURLOPT_POSTFIELDS => http_build_query($data)
+			);
+			$header = array(
+				'Host: ' . parse_url(self::URI, PHP_URL_HOST),
+				'Content-Type: application/x-www-form-urlencoded',
+				'Cookie: ' . self::FA_AUTH_COOKIE
+			);
+			return getSimpleHTMLDOM($this->getURI(), $header, $opts);
+	}
+
+	private function getFASimpleHTMLDOM($url, $cache = false) {
+			$header = array(
+				'Cookie: ' . self::FA_AUTH_COOKIE
+			);
+			if($cache) {
+				return getSimpleHTMLDOMCached($url, 86400, $header); // 24 hours
+			} else {
+				return getSimpleHTMLDOM($url, $header);
+			}
+	}
+
+	private function itemsFromJournalList($html, $limit) {
+		foreach($html->find('table[id^=jid:]') as $journal) {
+			if($limit-- === 0) break;
+
+			$item = array();
+
+			$this->cleanupHTMLDOM($journal);
+
+			$item['uri'] = $journal->find('a', 0)->href;
+			$item['title'] = html_entity_decode($journal->find('a', 0)->plaintext);
+			$item['author'] = $this->getInput('username-journals');
+			$item['timestamp'] = strtotime(
+				$journal->find('span.popup_date', 0)->plaintext);
+			$item['content'] = $journal
+				->find('.alt1 table div.no_overflow', 0)
+				->innertext;
+
+			$this->items[] = $item;
```

Comment:
Instead of `if($limit-- === 0) break;` above, you can do

```suggestion
			$this->items[] = $item;
			
			if (count($this->items) >= $limit) break;
```

Reply:
The reason this was done this way was to allow limit = -1 to mean "unlimited", like suggested in the relevant bridge parameters' titles. I added a comment about this in 2563019.

Maybe doing it the way you suggested and having limit = 9999 or such is more obvious for users though. What do you think?